src/components/UploadSection.svelte

```diff
@@ -14,6 +23,41 @@
         }
     });
 
+    async function sendFiles() {
+        for (let i: number = 0; i < files.length; i++) {
+            let exifData = null;
+
+            try {
+                let exifData = await exifr.parse(files[i]);
+            } catch (e) {
+                console.error(e);
+            }
+
+            let takenAtSec = null;
+
+            if (exifData) {
+                takenAtSec = exifData.DateTimeOriginal.valueOf();
+            }
+
+            let postFile: PostRequest = {
+                ObjectKey: files[i].name,
+                SizeBytes: files[i].size,
+                TakenAtSec: takenAtSec,
+            };
+
+            postRequest.push(postFile);
+        }
+        try {
+            postResponse = await postInsertRequest(postRequest);
+
+            for (let i = 0; i < files.length; i++) {
+                await pushMediaFilesToS3(postResponse[i].UploadUrl, files[i]);
+            }
```

Comment:
Usually, when you have await in for loop, that's a red flag.

We should wait for all uploads, so they can be in parallel.

See `Promise.all()`